tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+
+// Expected values
+console.log(input.value); // current (JointJS sets the 'value' DOM property internally)
+console.log(input.defaultValue); // initial
+console.log(input.getAttribute('value')); // initial
+</code></pre>
+
+        <p>
+            The supported properties when using the <code>props</code> attribute are:
+        </p>
+
+        <ul>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#checked" target="_blank"><code>checked</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option#attributes" target="_blank"><code>selected</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#disabled" target="_blank"><code>disabled</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#readonly" target="_blank"><code>readOnly</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable" target="_blank"><code>contentEditable</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#value" target="_blank"><code>value</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes" target="_blank"><code>indeterminate</code></a></li>
+            <li><a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#attributes" target="_blank"><code>multiple</code></a></li>
+        </ul>
+
+        <br>
+
+        <h3>Caveats!</h3>
+
+        <p>
+            While all that we have covered so far is great for users of JointJS, even in this day and age, it's not without some caveats.
+            Firstly, we will cover some minor points regarding syntax that you should be aware of when using HTML within your <code>markup</code>.
+            Afterwards, we'll look at some browser related quirks which mostly centre around Safari.
+        </p>
+
+
+        <h4>Syntax</h4>
+
+
+        <h4 style="font-weight: normal;">Boolean Attributes</h4>
+
+        <p>
+            Boolean attributes are those data types where the presence or absence of the attribute represents its <i>truthy</i> or <i>falsy</i>
+            value respectively. Some common examples are 
+            <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required" target="_blank"> 
+            <code>required</code></a>, 
+            <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled" target="_blank"><code>disabled</code></a>, 
+            and <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly" target="_blank"><code>readonly</code></a>.
+        </p>
+
+        <p>
+            Usually, boolean attributes can be utilized in 3 different ways, you can omit the value, use an empty string, or set the value as a
```

Comment:
I would use a colon or a hyphen instead of a comma here "...3 different ways: you can..."